include/libdnf/rpm/package_query.hpp

```diff
@@ -77,285 +77,495 @@ class PackageQuery : public PackageSet {
+        const PackageSet & package_set, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
 
-    /// cmp_type could be only libdnf::sack::QueryCmp::EQ, NEQ, GT, GTE, LT, LTE.
+    /// Filter packages by their `epoch:version-release`.
     ///
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_EPOCH
-    /// @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char **matches) - cmp_type = HY_PKG_EPOCH
-    PackageQuery & filter_epoch(
-        const std::vector<unsigned long> & patterns, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
+    /// @param patterns         A vector of strings the filter is matched against.
+    /// @param cmp              A comparison (match) operator, defaults to `QueryCmp::EQ`.
+    ///                         Supported values: `EQ`, `GT`, `LT`, `GTE`, `LTE`, `EQ`.
+    /// @since 5.0
+    //
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_EVR
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char **matches) - cmp_type = HY_PKG_EVR
+    PackageQuery & filter_evr(
+        const std::vector<std::string> & patterns, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
 
-    /// cmp_type could be only libdnf::sack::QueryCmp::EQ, NEQ, GLOB, NOT_GLOB.
-    PackageQuery & filter_epoch(
+    /// Filter packages by their `name-[epoch:]version-release.arch`. The following matches are tolerant to omited 0 epoch: `EQ`, `NEQ`, `GT`, `GTE`, `LT`, `LTE`.
+    ///
+    /// @param patterns         A vector of strings the filter is matched against.
+    /// @param cmp              A comparison (match) operator, defaults to `QueryCmp::EQ`.
+    ///                         Supported values: `EQ`, `NEQ`, `GT`, `GTE`, `LT`, `LTE`, `GLOB`, `NOT_GLOB`, `IGLOB`, `NOT_IGLOB`, `IEXACT`, `NOT_IEXACT`.
+    /// @since 5.0
+    //
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_NEVRA
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char **matches) - cmp_type = HY_PKG_NEVRA
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char *match) - cmp_type = HY_PKG_NEVRA_STRICT
+    // @replaces libdnf/sack/query.hpp:method:addFilter(int keyname, int cmp_type, const char **matches) - cmp_type = HY_PKG_NEVRA_STRICT
+    PackageQuery & filter_nevra(
         const std::vector<std::string> & patterns, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
 
-    /// cmp_type could be only libdnf::sack::QueryCmp::EQ, NEQ, GLOB, NOT_GLOB, IEXACT, NOT_IEXACT, ICONTAINS, NOT_ICONTAINS, IGLOB, NOT_IGLOB, CONTAINS, NOT_CONTAINS.
+    /// Filter packages by their `name-[epoch:]version-release.arch` based on values set in the `nevra` object.
```

Comment:
What about to replace `name-[epoch:]version-release.arch` by `name`, `epoch`, `version`, `release`, and `arch`?

Reply:
I intentionally used this to indicate what's the input format.
Individual values do not describe the input format at all.

Reply:
I believe that the input format is not relevant at all. You can set each string separately in nevra. You even don't need to parse string or parse string in NEVRA format.

Reply:
Right, this wasn't entirely correct because it's referring to nevra string rather than object attributes.
I've rephrased the doctext.